src/core/Menu/Menu.tsx

```diff
@@ -35,49 +47,34 @@ export const Menu = React.forwardRef<HTMLUListElement, MenuProps>(
     const refs = useMergedRefs(menuRef, ref);
 
     React.useEffect(() => {
-      const items = menuRef.current?.children;
+      const items = getFocusableItems(menuRef.current);
       if (focusedIndex != null) {
         (items?.[focusedIndex] as HTMLLIElement)?.focus();
         return;
       }
 
-      const childrenArray = React.Children.toArray(children);
-      const selectedIndex = childrenArray.findIndex(
-        (child: JSX.Element) => child.props.isSelected,
-      );
-      const firstEnabledIndex = childrenArray.findIndex(
-        (child: JSX.Element) => !child.props.disabled,
+      const selectedIndex = items.findIndex((el) =>
+        el.classList.contains('iui-active'),
       );
-      setFocusedIndex(selectedIndex > -1 ? selectedIndex : firstEnabledIndex);
+      setFocusedIndex(selectedIndex > -1 ? selectedIndex : 0);
     }, [children, focusedIndex]);
```

Comment:
Is children still needed here?

Reply:
Good catch! Moved to separate `useEffect`.